contrib/platform/src/com/sun/jna/platform/win32/Kernel32Util.java

```diff
@@ -678,4 +680,141 @@ public static final String extractVolumeGUID(String volumeGUIDPath) {
         
         return volumeGUIDPath.substring(VOLUME_GUID_PATH_PREFIX.length(), volumeGUIDPath.length() - VOLUME_GUID_PATH_SUFFIX.length());
     }
+
+	/**
+	 * Gets the specified resource out of the specified executable file
+	 * 
+	 * @param path
+	 *            The path to the executable file
+	 * @param type
+	 *            The type of the resource (either a type name or type ID is
+	 *            allowed)
+	 * @param name
+	 *            The name or ID of the resource
+	 * @return The resource bytes, or null if no such resource exists.
+	 */
+	public static byte[] getResource(String path, String type, String name) {
+		final HMODULE target = Kernel32.INSTANCE.LoadLibraryEx(path, null, Kernel32.LOAD_LIBRARY_AS_DATAFILE);
```

Comment:
As per the documentation in https://msdn.microsoft.com/en-us/library/windows/desktop/ms684179(v=vs.85).aspx:

 >If the function fails, the return value is NULL. To get extended error information, call GetLastError.

Your code should do this and then throw a _Win32Exception_ - see examples in _Kernel32Util_